src/js/tabular_comparison/indicator.js

```diff
@@ -0,0 +1,117 @@
+import React, {useState} from "react";
+import {Autocomplete, Box, Card, Grid, Icon, TextField} from "@mui/material";
+
+const Indicator = (props) => {
+    const [selectedIndicator, setSelectedIndicator] = useState(null);
+    const [categoryValue, setCategoryValue] = useState(null);
+
+    const indicatorAutoComplete = <Autocomplete
+        disabled={props.indicators.length <= 0}
+        isOptionEqualToValue={(option => true)}
+        options={[...new Set(props.indicators.map(item => item.indicator))]}
+        getOptionLabel={(option) => option}
+        onChange={(event, selectedValue) => handleIndicatorSelection(event, selectedValue)}
+        size={'small'}
+        renderOption={(props, option) => (
+            <Box
+                component="li"
+                {...props}
+            >
+                {option}
+            </Box>
+        )}
+        renderInput={(params) =>
+            <TextField
+                {...params}
+                label={'Indicator'}
+            />}
+    />
+
+    const categoryAutoComplete = <Autocomplete
+        disabled={selectedIndicator === null}
+        isOptionEqualToValue={(option => true)}
+        options={
+            selectedIndicator == null ? [] :
+                selectedIndicator.indicatorDetail.metadata.groups
+                    .filter((x) => x.name === selectedIndicator.indicatorDetail.metadata.primary_group)[0].subindicators
+        }
+        getOptionLabel={(option) => option}
+        size={'small'}
+        value={categoryValue}
+        onChange={(event, selectedValue) => handleCategorySelection(event, selectedValue)}
+        renderOption={(props, option) => (
+            <Box
+                component="li"
+                {...props}
+            >
+                {option}
+            </Box>
+        )}
+        renderInput={(params) =>
+            <TextField
+                {...params}
+                label={'Category'}
+            />}
+    />
+
+    const handleIndicatorSelection = (event, selectedValue) => {
+        let newValue = props.indicators.filter(x => x.indicator === selectedValue)[0];
+        setSelectedIndicator(newValue);
+        setCategoryValue(null);
+
+        props.handleIndicatorSelection(newValue);
+    }
+
+    const handleCategorySelection = (event, selectedValue) => {
+        setCategoryValue(selectedValue);
+        props.handleCategorySelection(selectedValue);
+    }
+
+    return (
+        <div>
+            <div
+                className={'remove-indicator'}
+                onClick={props.handleRemove}
+            >x
```

Comment:
Looks like x is a typo

Reply:
x is not a typo - it is there to create the remove button